Explorer/Assets/Scripts/ECS/StreamableLoading/DeferredLoading/DeferredLoadingSystem.cs

```diff
@@ -0,0 +1,91 @@
+using ECS.Abstract;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.AssetBundles;
+using ECS.StreamableLoading.AssetBundles.Manifest;
+using ECS.StreamableLoading.Common.Components;
+using ECS.StreamableLoading.Textures;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace ECS.Prioritization.DeferredLoading
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(PrepareAssetBundleManifestParametersSystem))]
+    [UpdateBefore(typeof(LoadAssetBundleManifestSystem))]
+    [UpdateBefore(typeof(LoadTextureSystem))]
+    [UpdateBefore(typeof(LoadAssetBundleSystem))]
+    public partial class DeferredLoadingSystem<TAsset, TIntention> : BaseUnityLoopSystem where TIntention: struct, ILoadingIntention
+    {
+        private static readonly QueryDescription CREATE_LOADING_REQUEST = new QueryDescription()
+                                                                         .WithAll<TIntention>()
+                                                                         .WithNone<LoadingInProgress, StreamableLoadingResult<TAsset>>();
+        private readonly Query query;
+
+        private readonly IConcurrentBudgetProvider concurrentLoadingBudgetProvider;
+        private readonly unsafe List<IntentionData> loadingIntentions;
+
+        public DeferredLoadingSystem(World world, ConcurrentLoadingBudgetProvider concurrentLoadingBudgetProvider) : base(world)
+        {
+            this.concurrentLoadingBudgetProvider = concurrentLoadingBudgetProvider;
+            query = World.Query(in CREATE_LOADING_REQUEST);
+            loadingIntentions = new List<IntentionData>();
+        }
+
+        protected override unsafe void Update(float t)
+        {
+            loadingIntentions.Clear();
+
+            foreach (ref Chunk chunk in query.GetChunkIterator())
+            {
+                ref TIntention intentionFirstElement = ref chunk.GetFirst<TIntention>();
+                ref PartitionComponent partitionFirstElement = ref chunk.GetFirst<PartitionComponent>();
+
+                foreach (int entityIndex in chunk)
+                {
+                    ref PartitionComponent partition = ref Unsafe.Add(ref partitionFirstElement, entityIndex);
+                    ref TIntention intention = ref Unsafe.Add(ref intentionFirstElement, entityIndex);
+
+                    void* intentionPointer = UnsafeUtility.AddressOf(ref Unsafe.Add(ref intentionFirstElement, entityIndex));
+
+                    var intentionData = new IntentionData()
+                    {
+                        PartitionComponent = partition,
+                        IntentionDataPointer = intentionPointer,
+                    };
+
+                    loadingIntentions.Add(intentionData);
+                }
+            }
+            loadingIntentions.Sort((p1, p2) => p1.PartitionComponent.CompareTo(p2.PartitionComponent));
```

Comment:
```suggestion
            loadingIntentions.Sort(static (p1, p2) => p1.PartitionComponent.CompareTo(p2.PartitionComponent));
```
Please place a `static` hint so it will be assured that no delegate is allocated every time at first sight.